### PR TITLE
Bump Typescript version to 5.3.3

### PR DESCRIPTION
**Pull request by marmoure**
Updates Typescript version and eslint, also fix linting error and code issues.

This PR has a commit for Adding linting to CI, Should we move it to another PR?

----------
Rebased on top of main on 19/04/2024

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -4062,10 +4062,25 @@ typedoc@^0.25.7:
     minimatch "^9.0.3"
     shiki "^0.14.7"
 
-typescript@4.6.3:
```
Comment (plutonik-a) [inline yarn.lock:4065]: This doesn't seem to be the matching yarn.lock file for this `typescript": "5.1.6"` update.

File: yarn.lock [hunk 2]
```diff
@@ -4062,25 +4062,10 @@ typedoc@^0.25.7:
...
-    commander "^2.20.0"
-    glob "^7.1.4"
-    is-absolute-url "^3.0.1"
-    is-html "^1.1.0"
-    jsdom "^14.1.0"
-    lodash "^4.17.15"
-    postcss "^7.0.17"
-    postcss-selector-parser "6.0.2"
-    request "^2.88.0"
+typescript@4.6.3:
```
Comment (plutonik-a) [inline yarn.lock]: Downgrade of `typescript@4.6.3` in yarn.lock doesn't match imported `"typescript": "5.3.3"` in package.json.

**Review by plutonik-a — CHANGES_REQUESTED** [review]
There seems to be a mismatch of typescript version in the history of committed `yarn.lock` files in contrast to the required typescript version declared in the according package.json, therefore it is impossible to reinstate and test previous single commits without changing the yarn lock file while installing the modules and breaking the build. 

The current build error indicates that there is a type compatibility issue in the saxes library, specifically in the `saxes.d.ts` file (type parameter `0` does not satisfy the constraint `SaxesOptions` and is not assignable to type `CommonOptions`).  
We need to fix the saxes version imported by JDita.

**Review by plutonik-a — APPROVED** [review]

File: yarn.lock (hunk 1 above)
Comment (marmoure) [reply yarn.lock:4065]: This issue happened while rebasing, `yarn.lock` had too many conflicts to and it was not clear which version to go with.
This conflicts rose after https://github.com/evolvedbinary/prosemirror-jdita/pull/241 was merged into main.
I made one commit to address the `yarn.lock` issue: [ee8a465](https://github.com/evolvedbinary/prosemirror-jdita/pull/233/commits/ee8a4659d01aea5070014e056c1cfbf1667751f1)

**Comment by marmoure** [comment]
Moved to #277

**Pull request closed without merging**